plugin/registereditor.lua

```diff
@@ -1,9 +1,136 @@
 local internals = require("internals")
 
+-- add a callback to a keypress without affecting existing keymaps
+local add_key_trigger = function(mode, key, callback, prepend)
+    -- get the current keymap for the key
+    local keymap = vim.fn.maparg(key, mode, false, true)
+
+    -- if there is no current keymap, create a new keymap with the new callback
+    if next(keymap) == nil then
+        vim.keymap.set(mode, key, callback)
+        return
+    end
+
+    -- create a new keymap that calls both the old and new callbacks
+    vim.keymap.set(mode, key, function()
+        local result
+        -- the ordering of the old and new callbacks can be chosen
+        if prepend then
+            callback()
+            result = keymap.callback()
+        else
+            result = keymap.callback()
+            callback()
+        end
+        return result
+    end, { expr = true, remap = true })
+end
```

Comment:
Many problems went over in https://github.com/tuurep/registereditor/pull/12#issuecomment-3000117395

One other problem: wouldn't work with builtin remaps like say:

```lua
vim.keymap.set("n", "<leader>/", "*")
vim.keymap.set("n", "<leader>?", "#")
```

(remapping builtins)

Reply:
If we only use this for register updating, we won't need the ability to prepend (but it's there for general purpose I see)